chacra/async/__init__.py

```diff
@@ -52,13 +52,14 @@ def poll_repos():
 
     """
     logger.info('polling repos....')
-    for r in models.Repo.query.filter_by(needs_update=True).all():
+    for r in models.Repo.query.filter_by(needs_update=True, is_queued=False).all():
         # this repo is being processed, do not pile up and try to get it
         # processed again until it is done doing work
         if r.is_updating:
             continue
         if r.needs_update:
             logger.info("repo %s needs to be updated/created", r)
+            r.is_queued = True
```

Comment:
you need to mark `r.is_queued = False` when a repo is done processing (in both rpm and deb queues)